genesis_notification/dm/models.py

```diff
@@ -0,0 +1,309 @@
+        types.String(min_length=1, max_length=128),
+        required=True,
+    )
+    port = properties.property(
+        types.Integer(),
+        required=True,
+    )
+    noreply_email_address = properties.property(
+        types.String(min_length=5, max_length=128),
+        required=True,
+    )
+
+    def _parse_message(self, message):
+        subject = ""
+        body = message
+        parts = message.split("\n", 3)
+        if len(parts) > 2 and parts[1] == "":
+            subject = parts[0]
+            body = parts[2]
+        return subject, body
+
+    def _build_message(self, message, user_context):
+        subject, body = self._parse_message(message)
+        msg = multipart.MIMEMultipart()
+        msg["From"] = self.noreply_email_address
+        msg["To"] = user_context["email"]
+        msg["Subject"] = subject
+        msg.attach(text.MIMEText(body))
+        return msg
+
+    def send(self, message, user_context):
+        msg = self._build_message(message, user_context)
+        with smtplib.SMTP(self.host, self.port) as smtp:
+            return smtp.sendmail(
+                from_addr=self.noreply_email_address,
+                to_addrs=user_context["email"],
+                msg=msg.as_string(),
+            )
+
+
+class Provider(
+    models.ModelWithUUID,
+    models.ModelWithRequiredNameDesc,
+    ModelWithAlwaysActiveStatus,
+    models.ModelWithProject,
+    models.ModelWithTimestamp,
+    orm.SQLStorableWithJSONFieldsMixin,
+):
+    __tablename__ = "proveders"
+    __jsonfields__ = ["protocol"]
+
+    protocol = properties.property(
+        types_dynamic.KindModelSelectorType(
+            types_dynamic.KindModelType(SimpleSmtpProtocol),
+        ),
+        required=True,
+    )
+
+    def send(self, message, user_context):
+        return self.protocol.send(
+            message=message,
+            user_context=user_context,
+        )
+
+
+class EventType(
+    models.ModelWithUUID,
+    models.ModelWithNameDesc,
+    ModelWithAlwaysActiveStatus,
+    models.ModelWithProject,
+    models.ModelWithTimestamp,
+    orm.SQLStorableMixin,
+):
+    __tablename__ = "event_types"
+
+
+class Template(
+    models.ModelWithUUID,
+    models.ModelWithRequiredNameDesc,
+    ModelWithAlwaysActiveStatus,
+    models.ModelWithProject,
+    models.ModelWithTimestamp,
+    orm.SQLStorableWithJSONFieldsMixin,
+):
+    __tablename__ = "templates"
+    __jsonfields__ = ["params"]
+
+    content = properties.property(
+        types.String(max_length=10240),
+        required=True,
+    )
+    params = properties.property(
+        types.Dict(),
+        required=True,
+    )
+    provider = relationships.relationship(
+        Provider,
+        required=True,
+        prefetch=True,
+    )
+    event_type = relationships.relationship(EventType, required=True)
+    is_default = properties.property(
+        types.Boolean(),
+        default=False,
+    )
+
+
+class UserExcenge(types_dynamic.AbstractKindModel):
```

Comment:
```suggestion
class UserExchange(types_dynamic.AbstractKindModel):
```

and in other places too

Reply:
done